server/store/store.go

```diff
@@ -403,72 +408,125 @@ func (s *SQLStore) MattermostToTeamsUserID(userID string) (string, error) {
 }
 
 func (s *SQLStore) GetPostInfoByMSTeamsID(chatID string, postID string) (*storemodels.PostInfo, error) {
-	query := s.getQueryBuilder().Select("mmPostID, msTeamsLastUpdateAt").From("msteamssync_posts").Where(sq.Eq{"msTeamsPostID": postID, "msTeamsChannelID": chatID})
+	tx, err := s.BeginTx()
+	if err != nil {
+		return nil, err
+	}
+
+	var txErr error
+	defer func() {
+		if txErr != nil {
+			if err := s.RollbackTx(tx); err != nil {
+				s.api.LogWarn("Unable to rollback database transaction", "error", err.Error())
+			}
+			return
+		}
+
+		if err := s.CommitTx(tx); err != nil {
+			s.api.LogWarn("Unable to commit database transaction", "error", err.Error())
+		}
+	}()
+
+	query := s.getQueryBuilder().Select("mmPostID, msTeamsLastUpdateAt").From("msteamssync_posts").Where(sq.Eq{"msTeamsPostID": postID, "msTeamsChannelID": chatID}).Suffix("FOR UPDATE").RunWith(tx)
 	row := query.QueryRow()
 	var lastUpdateAt int64
 	postInfo := storemodels.PostInfo{
 		MSTeamsID:      postID,
 		MSTeamsChannel: chatID,
 	}
-	err := row.Scan(&postInfo.MattermostID, &lastUpdateAt)
-	if err != nil {
-		return nil, err
+
+	if txErr = row.Scan(&postInfo.MattermostID, &lastUpdateAt); txErr != nil {
+		return nil, txErr
 	}
 	postInfo.MSTeamsLastUpdateAt = time.UnixMicro(lastUpdateAt)
 	return &postInfo, nil
 }
 
 func (s *SQLStore) GetPostInfoByMattermostID(postID string) (*storemodels.PostInfo, error) {
-	query := s.getQueryBuilder().Select("msTeamsPostID, msTeamsChannelID, msTeamsLastUpdateAt").From("msteamssync_posts").Where(sq.Eq{"mmPostID": postID})
+	tx, err := s.BeginTx()
```

Comment:
Same here, and probably most of the cases in this file. The transaction is only needed when you operate with multiple queries, and any of those queries is a modification query.